src/app/core/params.ts

```diff
@@ -0,0 +1,215 @@
+        return new Error("Parameter " + key + " is the wrong type. " + "Expected: " + expected + ", Actual: " + actual);
+    }
+    /**
+     * Turns the raw key:value pairs passed into a user-written Feature declaration
+    into key:Parameter pairs. This forces the checks for each Parameter type
+    to execute on the provided values, and should throw an error for mismatches.
+     * @param {*} values 
+     * @returns {Parameters} Returns a parameters object
+     * @memberof Params
+     */
+    __sanitizeValues(values: any) {
+        /*let newParams: any = {};
+        for (let key in values) {
+            let oldParam = values[key];
+            if (this.isUnique(key)) {
+                newParams[key] = Parameter.makeParam(this.unique[key], oldParam);
+            } else if (this.isHeritable(key)) {
+                if (values[key]) {
+                    newParams[key] = Parameter.makeParam(this.heritable[key], oldParam);
+                }
+            } else {
+                throw new Error(key + " does not exist in this set of ParamTypes: " + Object.keys(this.unique) + Object.keys(this.heritable));
+            }
+        }
+        this.__checkParams(newParams);
+        return newParams;
+        */
+       throw new Error("Sanitize values inoperable.");
+    }
+
+    /* Checks to make sure the set of sanitized parameters matches the expected ParamTypes.
+    This method also checks to make sure that all unique (required) params are present.*/
+    /**
+     * Checks to make sure the set of sanitized parameters matches the expected ParamTypes.
+    This method also checks to make sure that all unique (required) params are present.
+     * @param {Params} parameters 
+     * @memberof Params
+     * @returns {void}
+     */
+    __checkParams(parameters: {[index: string]: Parameter}) {
+        for (let key in parameters) {
+            let param = parameters[key];
+            if (!(param instanceof Parameter)) {
+                throw new Error(key + " is not a ParameterValue.");
+            } else if (this.isUnique(key)) {
+                /*if (param.__type != this.unique[key]) {
+                    this.wrongTypeError(key, this.unique[key], param.__type);
+                }*/
+                throw new Error("Unique check inoperable");
+            } else if (this.isHeritable(key)) {
+                /*if (param.__type != this.heritable[key]) {
+                    this.wrongTypeError(key, this.heritable[key], param.__type);
+                }*/
+                throw new Error("Heritable check inoperable");
+            } else {
+                throw new Error(key + " does not exist in this set of ParamTypes.");
+            }
+        }
+        if (!this.hasAllUniques(parameters)) {
+            throw new Error("Unique values were not present in the provided parameters. Expected: " + Object.keys(this.unique) + ", saw: " + Object.keys(parameters));
+        }
```

Comment:
Similar problem, we will need to move these two functions elsewhere to simplify the debugging of the parameters errors.